bls/bls.go

```diff
@@ -40,3 +41,25 @@ func Verify(pubkey, msg, sig []byte) bool {
 
 	return true
 }
+
+// GetPubKey returns the public key from the private key.
+func GetPubKey(pk []byte) (address.Address, error) {
+	var pkrev [32]byte
+	for i := 0; i < 32; i++ {
+		pkrev[i] = pk[32-i-1]
+	}
+	scalar := curve12381.NewKyberScalar().SetBytes(pkrev[:])
+	pubPoint := curve12381.NewGroupG1().Point()
+	pubPoint.Mul(scalar, nil)
```

Comment:
As defined in the Filecoin spec:
> Filecoin currently uses the group G_1 for representing public keys and the group G_2 for representing signatures.

And per BLS RFC:
> PK = x*P2 (where x is the private key scalar)

The BLS library considers the `nil` point in G1 as the base point of the group:
```
// Multiply point p by the scalar s.                                                                                                                                                                                                                                            
// If p == nil, multiply with the standard base point Base().                                                                                                                                                                                                                   
Mul(s Scalar, p Point) Point
```